src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityService.kt

```diff
@@ -0,0 +1,74 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationListener
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.util.UriComponentsBuilder
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEvent
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEventType
+import java.time.LocalDate
+
+@Service
+class CommunityService(
+  @Value("\${communityapi.temporaryReferralRequest.probationAreaCode}") private val probationAreaCode: String,
+  @Value("\${communityapi.temporaryReferralRequest.referralType}") private val referralType: String,
+  @Value("\${communityapi.temporaryReferralRequest.staffCode}") private val staffCode: String,
+  @Value("\${communityapi.temporaryReferralRequest.teamCode}") private val teamCode: String,
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.sent-referral}") private val interventionsUISentReferralLocation: String,
+  @Value("\${community-api.locations.sent-referral}") private val communityAPISentReferralLocation: String,
+  private val communityApiWebClient: WebClient,
+) : ApplicationListener<ReferralEvent> {
+
+  override fun onApplicationEvent(event: ReferralEvent) {
+    when (event.type) {
+      ReferralEventType.SENT -> {
+        val url = UriComponentsBuilder.fromHttpUrl(interventionsUIBaseURL)
+          .path(interventionsUISentReferralLocation)
+          .buildAndExpand(event.referral.id)
+          .toString()
+
+        val body = ReferRequest(
+          probationAreaCode,
+          referralType,
+          staffCode,
+          teamCode,
+          url,
+          event.referral.sentAt!!.toLocalDate()
+        )
+
+        val communityApiSentReferralPath = UriComponentsBuilder.fromPath(communityAPISentReferralLocation)
+          .buildAndExpand(event.referral.serviceUserCRN)
+          .toString()
+
+        communityApiWebClient.post().uri(communityApiSentReferralPath)
+          .body(Mono.just(body), ReferRequest::class.java)
+          .retrieve()
+          .bodyToMono(Unit::class.java)
+          .onErrorResume { e ->
+            log.error("Call to community api to update contact log failed: {}", e.toString())
```

Comment:
sorry missed this before - the second argument here should be the error itself, `e`.